src/sentry/models/dashboard_widget.py

```diff
@@ -114,6 +118,61 @@ class Meta:
     __repr__ = sane_repr("widget", "type", "name")
 
 
+class DashboardWidgetQueryOnDemandManager(Manager):
+    def filter_enabled(self):
+        return super().get_queryset().filter(extraction_reason__startswith=_ON_DEMAND_ENABLED_KEY)
+
+
+@region_silo_only_model
+class DashboardWidgetQueryOnDemand(Model):
+    """
+    Tracks on_demand state and values for dashboard widget queries.
+    Only a subset of dashboard widget queries have conditions or columns that would
+    require on-demand extraction, and others are simply not applicable (eg. different dataset).
+    """
+
+    __relocation_scope__ = RelocationScope.Organization
+
+    dashboard_widget_query = FlexibleForeignKey("sentry.DashboardWidgetQuery")
+
+    spec_hashes = ArrayField()
+
+    class OnDemandExtractionState(models.TextChoices):
+        DISABLED_NOT_APPLICABLE = "disabled:not_applicable", gettext_lazy("disabled:not_applicable")
+        """ This widget does not have on-demand metrics needing extraction. """
+        DISABLED_PREROLLOUT = "disabled:pre_rollout", gettext_lazy("disabled:pre_rollout")
+        """ This represents a pre-filled on-demand value to do load estimates before enabling extraction. """
+        DISABLED_MANUAL = "disabled:manual", gettext_lazy("disabled:manual")
+        """ The widget was manually disabled by a user """
+        DISABLED_SPEC_LIMIT = "disabled:spec-limit", gettext_lazy("disabled:spec-limit")
+        """ This widget query was disabled during rollout due to the organization reaching it's spec limit. """
+        DISABLED_HIGH_CARDINALITY = "disabled:high-cardinality", gettext_lazy(
+            "disabled:high-cardinality"
+        )
+        """ This widget query was disabled by the cardinality cron due to one of the columns having high cardinality """
+        ENABLED_ROLLED_IN = "enabled:rolled_in", gettext_lazy("enabled:rolled_in")
```

Comment:
```suggestion
        ENABLED_ENROLLED = "enabled:enrolled", gettext_lazy("enabled:enrolled")
```

Would enrolled be the right term? I can't find "rolled-in" used in this way.